core/meta/test/testInterpreter.cxx

```diff
@@ -0,0 +1,11 @@
+#include "TInterpreter.h"
+#include "TSystem.h"
+
+#include "gtest/gtest.h"
+#include "gmock/gmock.h"
+
+TEST(TInterpreter, ErrnoValue)
+{
+   gInterpreter->ProcessLine("errno");
```

Comment:
Do you expect this to reset `errno` to 0? If not, why would you expect `TSystem::Errno()` to return 0 below?

Reply:
I expect the invocation not to change the value of errno.
Presently it does and this is the cause of the bug I am trying to fix, even if the issue was reported through an example involving TFile.
I realise that perhaps this "recursion" (it must not be "errno", it could be "TFile f;") might be confusing but it is stand-alone, a peculiarity which is perhaps desirable for tests in core, right?

Reply:
Sorry, can't follow. If something sets `errno` to 7, then calls the `ErrnoValue` test function then even if cling were to not touch `errno` the test would fail, correct?

I.e. I believe the test
- needs to reset `errno` to 0
- needs to trigger cling to override the previous `errno` (in the current master, fixed by your patch)

And in principle, if the test sets `errno` to 7 instead of 0 then after cling does its thing it should be 7 again - i.e. cling will need to set it to the previous value, not 0.

That shows that `errno` is a terrible concept: users must guarantee that nothing touches it in between setting and reading, or they will get an unexpected value. But that's just the usual global mess. Is this really worth fixing? I'm afraid it will be a never ending chase, especially because we must in principle also retain `errno` values != 0, i.e. any writing to log files, opening of scripts and headers etc will affect `errno`.

I'd rather put a big warning into the doc of `TFile::GetErrno()` - "this likely doesn't help you".

Reply:
I agree. It's a hard one to fix properly given the meaning of `TSystem::GetErrno()` and, ultimately, errno. 
Indeed a RAII could be put instead of the simple line in the autoparsing procedure but still it will be fragile...